L/libjulia/common.jl

```diff
@@ -0,0 +1,162 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg.Types
+
+# Collection of sources required to build GMP
+function configure(version)
+    name = "libjulia"
+
+    checksums = Dict(
+        v"1.4.2" => "76a94e06e68fb99822e0876a37c2ed3873e9061e895ab826fd8c9fc7e2f52795",
+        v"1.5.1" => "1f138205772eb1e565f1d7ccd6f237be8a4d18713a3466e3b8d3a6aad6483fd9",
+    )
+    sources = [
+        ArchiveSource("https://github.com/JuliaLang/julia/releases/download/v$(version)/julia-$(version).tar.gz", checksums[version]),
+        DirectorySource("./bundled"),
+    ]
+
+    # Bash recipe for building across all platforms
+    script = raw"""
+    apk update
+    apk add coreutils libuv-dev utf8proc
+
+    cd $WORKSPACE/srcdir/julia*
+
+    # Apply patches
+    if [ -d $WORKSPACE/srcdir/patches ]; then
+    for f in $WORKSPACE/srcdir/patches/*.patch; do
+        echo "Applying path ${f}"
+        atomic_patch -p1 ${f}
+    done
+    fi
+
+    case ${target} in
+        *linux*)
+            OS=Linux
+        ;;
+        *mingw*)
+            OS=WINNT
+        ;;
+        *darwin*)
+            OS=Darwin
+        ;;
+        *freebsd*)
+            OS=FreeBSD
+        ;;
+    esac
+
+    cat << EOM >Make.host.user
+    override CC=/opt/x86_64-linux-musl/bin/x86_64-linux-musl-gcc
+    override CXX=/opt/x86_64-linux-musl/bin/x86_64-linux-musl-g++
+    override AR=/opt/x86_64-linux-musl/bin/x86_64-linux-musl-ar
```

Comment:
Just to make this a bit more flexible:
```suggestion
    override CC=${CC_BUILD}
    override CXX=${CXX_BUILD}
    override AR=${AR_BUILD}
```